client/ayon_blender/plugins/publish/extract_camera_fbx.py

```diff
@@ -49,8 +49,22 @@ def process(self, instance):
         context = plugin.create_blender_context(
             active=camera, selected=selected)
 
-        scale_length = bpy.context.scene.unit_settings.scale_length
-        bpy.context.scene.unit_settings.scale_length = 0.01
+        scene = bpy.context.scene
+        scale_length = scene.unit_settings.scale_length
+        frame_start = scene.frame_start
+        frame_end = scene.frame_end
+        frame_step = scene.frame_step
+        fps = scene.render.fps
+        fps_base = scene.render.fps_base
+        scene.unit_settings.scale_length = instance.data.get(
+            "unitScale", scale_length)
+        scene.frame_start = instance.data.get("frameStart", frame_start)
+        scene.frame_end = instance.data.get("frameEnd", frame_end)
+        scene.frame_step = instance.data.get("frameStep", frame_step)
+        inst_fps = instance.data.get("fps")
```

Comment:
Shouldn't this just use the `global_scale` argument of the export function? [see here](https://docs.blender.org/api/current/bpy.ops.export_scene.html#module-bpy.ops.export_scene) or why is changing the scene better?

If we are to change the scene - then this should be done via a context manager!

> agreed. This should be done with the contextmanager + context.exitstack()
> You can get some inspirations here: https://github.com/ynput/ayon-core/pull/674#discussion_r1649068696


See: https://github.com/ynput/ayon-core/pull/658#discussion_r1644015067

Reply:
The changing of the scene scale length for me makes absolutely no difference on the exported FBX? Also, the "Unit scale" value in the publisher UI is the actual unit scale - and not a _scale factor_ to your scene units... which may be confusing as well.

Why wouldn't we use the `global_scale` instead as scaling factor?

Reply:
I think we should use the global scale instead and remove the related setting inside the Publisher UI